TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TwoControllerDriverControlled.java

```diff
@@ -74,40 +65,42 @@ public void runOpMode() throws InterruptedException {
 
-                //Applies calculated movement for both field-centric and not
-                if (fieldCentric) {
-                    //robot.getHeading() may be partially or not at all functional. Good luck, traveler.
-                    robot.drive.calculateDirectionsFieldCentric(x, y, r, robot.getHeading());
+                //employs gyro stabilization to linear driving
+                if (Math.abs(gamepad1.right_stick_x) < 0.05) {
+                    if (gyroAssistDebouncer.isPressed(Math.abs(gamepad1.right_stick_y) < 0.05)) {
+                        //identifies the desired heading angle to hold. This should only happen once per period of time that the right joystick is < 0.05
+                        target = robot.imu.getRobotYawPitchRollAngles().getYaw(AngleUnit.DEGREES);
+                    }
+                    robot.drive.calcGyroStabilized(x, y, target);
                 }
                 else {
-                    //applies drive values. Notice the negative R.
                     robot.drive.calculateDirections(x, y, r);
                 }
-                //Toggle field centric mode
-                //First implementation of new debouncer. If this fails at competition just delete.
-                if (dx.isPressed(gamepad1.triangle || gamepad2.x)) {
-                    fieldCentric = !fieldCentric;
-                }
-
-                //180
-                if (turnAroundDebouncer.isPressed(gamepad1.b)) {
-                    robot.drive.rotateLeftEncoder(180);
+                //90 degree turn left/right
+                if (gamepad1.left_trigger > 0.5) {
+                    if (turnAroundDebouncer.isPressed(gamepad1.left_trigger > 0.5)){
+                        target = robot.imu.getRobotYawPitchRollAngles().getYaw(AngleUnit.DEGREES) - 90;
+                    }
+                    robot.drive.calcGyroStabilized(0, 0, target);
                 }
-
-                //90
-                if (turnAroundDebouncer.isPressed(gamepad1.left_trigger > .5)) {
-                    tempTarget = robot.imu.getRobotYawPitchRollAngles().getYaw(AngleUnit.DEGREES)-90;
-                    while (turnAroundDebouncer.isPressed(gamepad1.left_trigger > .5)) {
-                        robot.drive.calcGryoStabilized(0,0, tempTarget);
-                        robot.drive.applyPower();
+                if (gamepad1.right_trigger > 0.5) {
+                    if (turnAroundDebouncer.isPressed(gamepad1.right_trigger > 0.5)){
+                        target = robot.imu.getRobotYawPitchRollAngles().getYaw(AngleUnit.DEGREES) + 90;
                     }
+                    robot.drive.calcGyroStabilized(0, 0, target);
                 }
-                if (turnAroundDebouncer.isPressed(gamepad1.right_trigger > 0.5)) {
-                    robot.drive.calcGryoStabilized(0,0, robot.imu.getRobotYawPitchRollAngles().getYaw(AngleUnit.DEGREES)+90);
+                //180 degree turn
+                if (gamepad1.b) {
```

Comment:
this top `gamepad1.b` is superflous. the bottom debouncer is all that is needed.

Reply:
No, it has a purpose actually. There are some functions that need to be run only once when the button is pressed, and others that need to be run continuously for as long as the button is pressed.